test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/ReadersTest.scala

```diff
@@ -364,3 +364,10 @@ class InputStreamReaderTest {
     assertThrows(classOf[IOException], r.mark(0))
   }
 }
+
+class FilterReaderTest {
```

Comment:
This is a bit light, as far as tests goes. Could we at least test that giving it an actual reader as argument allows to call methods of that reader through the filter proxy?

Reply:
Also, consider putting this in a dedicated file. I know the current file bundles a lot of them, but that's not great, and we should try not to perpetuate past mistakes. :sweat_smile:

Reply:
I was sort of expecting the nice reused tests like in Collections :-)

I can put the others in separate files as well if you like?

Reply:
> I can put the others in separate files as well if you like?

If you feel like it, why not :)